pybossa/auth/project.py

```diff
@@ -51,11 +51,14 @@ def _create(self, user, project=None):
         return self.only_admin_or_subadmin(user)
 
     def _read(self, user, project=None):
-        if project is not None and project.published is False:
-            return self.only_admin_or_subadminowner(user, project)
-        if project is not None and data_access.data_access_levels:
-            return self.only_admin_or_subadminowner(user, project) or \
-                self.only_project_users(user, project)
+        if project:
+            if project.published is False:
+                return self.only_admin_or_subadminowner(user, project)
+            # private gigwork
+            if data_access.data_access_levels or \
+                (not data_access.data_access_levels and len(project.get_project_users()) > 0): # public gigwork and no users assigned to project
+                return self.only_admin_or_subadminowner(user, project) or \
+                    self.only_project_users(user, project)
         return user.is_authenticated
```

Comment:
See if its possible to simplify the code further; something similar to below
```python
if not project:
    return abort(404)
if project.published is False:
    return self.only_admin_or_subadminowner(user, project)
has_data_access = data_access.data_access_levels
has_project_users = len(project.get_project_users()) > 0
if has_data_access or (not has_data_access and has_project_users):
    return self.only_admin_or_subadminowner(user, project) or self.only_project_users(user, project)
return user.is_authenticated
```

Reply:
Refactored. However, I don't think `abort` statement belongs in the `Auth` class.